keras_core/backend/jax/distribution.py

```diff
@@ -0,0 +1,117 @@
+"""!!!DO NOT USE!!!
+
+Distribution related class for JAX backend.
+
+This is just a prototype and we might want to unify it in future for other
+backends.
+"""
+
+import contextlib
+
+from absl import logging
+import jax
+import numpy as np
+
+from keras_core.backend.common import global_state
+
+_DEFAULT_BATCH_DIM_NAME = "batch"
+_GLOBAL_ATTRIBUTE_NAME = "distribution"
+
+
+def get_distribution():
+    """Retrieve the current distribution from global context."""
+    return global_state.get_global_attribute(_GLOBAL_ATTRIBUTE_NAME)
+
+
+class DataParallelDistribution:
```

Comment:
So this is for data parallel training only? What would the full set of classes look like?

- DataParallelDistribution
- ModelParallelDistribution
- DataAndModelParallelDistribution

Is it worth considering just one class (the data and model parallelism class) that does all of this, and just make it easy to configure it for data parallelism? E.g. pass a 1d mesh, or a list of devices with no other args, you get all data parallelism.

No idea if that's a good idea, just trying to grok where we are heading...

Reply:
Thanks for the question, I think eventually there will be one base class that host most of the functionality, and DataParallelDistribution/ModelParallelDistribution will be the subclass for it. For now I haven't decide the final naming for the API yet, but I think ModelParallelDistribution will be the next one.